editoast/src/views/timetable.rs

```diff
@@ -126,21 +130,40 @@ async fn get(
     State(db_pool): State<DbConnectionPoolV2>,
     Extension(auth): AuthenticationExt,
     Path(TimetableIdParam { id: timetable_id }): Path<TimetableIdParam>,
-) -> Result<Json<TimetableDetailedResult>> {
+    Query(pagination_params): Query<PaginationQueryParams>,
+) -> Result<Json<ListTrainSchedulesResponse>>  {
     let authorized = auth
-        .check_roles([BuiltinRole::TimetableRead].into())
-        .await
-        .map_err(AuthorizationError::AuthError)?;
+    .check_roles([BuiltinRole::TimetableRead].into())
+    .await
+    .map_err(AuthorizationError::AuthError)?;
     if !authorized {
         return Err(AuthorizationError::Forbidden.into());
     }
 
     let conn = &mut db_pool.get().await?;
-    let timetable = TimetableWithTrains::retrieve_or_fail(conn, timetable_id, || {
+    Timetable::retrieve_or_fail(conn, timetable_id, || {
         TimetableError::NotFound { timetable_id }
     })
     .await?;
-    Ok(Json(timetable.into()))
+
+    let settings = pagination_params
+    .validate(25)?
+    .into_selection_settings()
+    .filter(move || TrainSchedule::TIMETABLE_ID.eq(timetable_id));
+
+    let (train_schedules, stats) = TrainSchedule::list_paginated(conn, settings).await?;
+    let results = train_schedules.into_iter().map_into().collect();
+
+    Ok(Json(ListTrainSchedulesResponse {stats, results}))
+}
+
+#[derive(Serialize, ToSchema)]
+#[cfg_attr(test, derive(Deserialize))]
+struct ListTrainSchedulesResponse {
+    #[schema(value_type = Vec<TrainScheduleResult>)]
+    results: Vec<TrainScheduleResult>,
+    #[serde(flatten)]
+    stats: PaginationStats,
 }
```

Comment:
nitpick: how about moving the `struct` above the function where it is used (I believe I can usually expect to find the `struct` used in a function by looking before the function).